### PR TITLE
NavigationService updates

### PR DESCRIPTION
**Pull request by mvegaca**
Related to #1012 
Add navigation events in navigation service to templates.
Reviewed and passed al tests

### CODE REVIEW

**Comment by crutkas** [comment]
Can we break the sample out from the template code update PR?  it is hard to break the two apart and i think it is critical to review the template updates.

It is also unclear why both #1012 and #775 both refer to documentation updates yet this has actual template updates.

File: templates/_composition/CodeBehind/Project/Services/NavigationService.cs [hunk 1]
```diff
@@ -51,5 +60,24 @@ public static bool Navigate(Type pageType, object parameter = null, NavigationTr
         public static bool Navigate<T>(object parameter = null, NavigationTransitionInfo infoOverride = null)
             where T : Page
             => Navigate(typeof(T), parameter, infoOverride);
+        
+        private static void RegisterFrameEvents()
+        {
+            _frame.Navigated += _frame_Navigated;
+            _frame.NavigationFailed += _frame_NavigationFailed;
+            _frameEventsRegistrated = true;
```
Comment (crutkas) [inline templates/_composition/CodeBehind/Project/Services/NavigationService.cs]: if we set this here, why isn't it unset in unregister?

File: templates/_composition/MVVMBasic/Project/Services/NavigationService.cs [hunk 2]
```diff
@@ -51,5 +59,24 @@ public static bool Navigate(Type pageType, object parameter = null, NavigationTr
         public static bool Navigate<T>(object parameter = null, NavigationTransitionInfo infoOverride = null)
             where T : Page
             => Navigate(typeof(T), parameter, infoOverride);
+        
+        private static void RegisterFrameEvents()
+        {
+            _frame.Navigated += _frame_Navigated;
+            _frame.NavigationFailed += _frame_NavigationFailed;
+            _frameEventsRegistrated = true;
```
Comment (crutkas) [inline templates/_composition/MVVMBasic/Project/Services/NavigationService.cs]: if we set this here, why isn't this unset in unregister

File: templates/_composition/MVVMLight/Project/Services/NavigationServiceEx.cs [hunk 3]
```diff
@@ -85,5 +93,24 @@ public string GetNameOfRegisteredPage(Type page)
                 }
             }
         }
+
+        private void RegisterFrameEvents()
+        {
+            _frame.Navigated += _frame_Navigated;
+            _frame.NavigationFailed += _frame_NavigationFailed;
+            _frameEventsRegistrated = true;
```
Comment (crutkas) [inline templates/_composition/MVVMLight/Project/Services/NavigationServiceEx.cs]: if we set this here, why isn't this unset in unregister

**Comment by crutkas** [comment]
Thanks for breaking this out, why do we set but never unset the bool for reg'ing the events?

Why do we even care, best of my knowledge, you can remove events that were never wired up with no issue.

**Comment by mvegaca** [comment]
Your are right @crutkas thank you very much.

**Comment by crutkas** [comment]
Why do we still have that bool check.  It *could* double up registering the events but you check to unreg?  wouldn't you check if the events are already wired as well?

**Comment by mvegaca** [comment]
You are right @crutkas it was enough with frame not null validation

**Pull request merged**